From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 8 Apr 2020 16:25:03 +0200
Subject: units: make sure systemd-pstore stops at shutdown

This doesn't matter too much given that the service doesn't do anything
on shutdown, but let's still stop it to make things cleaner.

(cherry picked from commit b0c1a07654c80d3cbbbcc52f860d4206707c0b08)
---
 units/systemd-pstore.service.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/units/systemd-pstore.service.in b/units/systemd-pstore.service.in
index f9705b5..6e4827f 100644
--- a/units/systemd-pstore.service.in
+++ b/units/systemd-pstore.service.in
@@ -13,7 +13,8 @@ Documentation=man:systemd-pstore(8)
 ConditionDirectoryNotEmpty=/sys/fs/pstore
 ConditionVirtualization=!container
 DefaultDependencies=no
-Before=sysinit.target
+Conflicts=shutdown.target
+Before=sysinit.target shutdown.target
 
 [Service]
 Type=oneshot
